libvast/vast/defaults.hpp

```diff
@@ -27,53 +25,52 @@ namespace vast::defaults {
 namespace command {
 
 /// Path to persistent state.
-extern std::string_view directory;
+constexpr std::string_view directory = "vast";
 
 /// Hostname or IP address of a remote node.
-extern std::string_view endpoint_host;
+constexpr std::string_view endpoint_host = "";
 
 /// TCP port address of a remote node.
-extern uint16_t endpoint_port;
+constexpr uint16_t endpoint_port = 42000;
 
 /// Server ID for the consensus module.
-extern std::string_view id;
+constexpr std::string_view id = "";
 
 /// Path for reading input events or `-` for reading from STDIN.
-extern std::string_view read_path;
+constexpr std::string_view read_path = "-";
 
 /// Path to alternate schema.
-extern std::string_view schema_path;
 
 /// Path for writing query results or `-` for writing to STDOUT.
-extern std::string_view write_path;
+constexpr std::string_view write_path = "-";
 
 /// Inverse factor by which to delay packets. For example, if 5, then for two
 /// packets spaced *t* seconds apart, the source will sleep for *t/5* seconds.
-extern int64_t pseudo_realtime_factor;
+constexpr int64_t pseudo_realtime_factor = 0;
 
 /// Number of bytes to keep per event.
-extern size_t cutoff;
+constexpr size_t cutoff = std::numeric_limits<size_t>::max();
 
 /// Flow table expiration interval.
-extern size_t flow_expiry;
+constexpr size_t flow_expiry = 10;
 
 /// Flush to disk after that many packets.
-extern size_t flush_interval;
+constexpr size_t flush_interval = 10000;
 
 /// Maximum number of results.
-extern size_t max_events;
+constexpr size_t max_events = 0;
 
 /// Maximum flow lifetime before eviction.
-extern size_t max_flow_age;
+constexpr size_t max_flow_age = 60;
 
 /// Number of concurrent flows to track.
-extern size_t max_flows;
+constexpr size_t max_flows = 1 << 20;
```

Comment:
While we're changing this: can we use SI literals here for improved clarity?

Reply:
It used to be defined with SI literals in the `.cpp`. I changed it for two reasons:
1. We should not do `using namespace ...` in the header, and when I have to choose between `vast::si_literals::operator""_Mi(1)` and `1 << 20` I choose the latter.
2. Pulling in an extra header for a single trivial symbol feels wasteful.

Reply:
I see the namespace argument, but I find `1 << 20` to cryptic for someone who might not be a seasoned coder. Maybe we can write it out instead and use `'` delimiters?